File: C# (ASP.NET & WPF)/C# - INSY7315 - EliteRentals - (Web App, Mobile App & API)/EliteRentals-WebApp/Models/DTOs/MessageDto.cs
using System;

namespace EliteRentals.Models.DTOs
{
    public class MessageDto
    {
        public int MessageId { get; set; }
        public int SenderId { get; set; }
        public string SenderName { get; set; } = "";
        public int? ReceiverId { get; set; }
        public string ReceiverName { get; set; } = "";
        public string MessageText { get; set; } = "";
        public DateTime Timestamp { get; set; }
        public bool IsChatbot { get; set; } = false;
        public bool IsBroadcast { get; set; }
        public string? TargetRole { get; set; }

        public bool IsRead { get; set; }

        public DateTime? ArchivedDate { get; set; }

    }
}

File: C# (ASP.NET & WPF)/C# - INSY7315 - EliteRentals - (Web App, Mobile App & API)/Elite-Rentals-Mobile-App/app/src/main/java/com/rentals/eliterentals/MessageDto.kt
package com.rentals.eliterentals

data class MessageDto(
    val messageId: Int? = null,
    val senderId: Int,
    val receiverId: Int? =null,
    val messageText: String,
    val timestamp: String? = null,
    val isChatbot: Boolean = false,
    val isBroadcast: Boolean = false,
    val targetRole: String? = null,
    val isEscalated: Boolean = false,
    val language: String? = null
)


